projects/pastanaga/src/lib/calendar/date-input.component.ts

```diff
@@ -0,0 +1,70 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, EventEmitter, Input, Output } from '@angular/core';
+import { markForCheck } from '../common/utils';
+import { TextfieldCommon } from '../textfield/textfield.common';
+import { format, isValid, isToday, isYesterday, getMonth } from 'date-fns';
+
+@Component({
+    selector: 'pa-date-input',
+    templateUrl: './date-input.component.html',
+    styleUrls: ['../popup/_popup.scss', './date-input.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DateInputComponent extends TextfieldCommon {
+
+    dateInput = '';
+    isValidDate = true;
+    currentDate: Date = new Date();
+    focused = false;
```

Comment:
as a convention, inner properties like these one should be after `@input` and `@output`